yt_dlp/downloader/http.py

```diff
@@ -99,9 +105,13 @@ def establish_connection():
             else:
                 range_start = None
             ctx.is_resume = False
-            range_end = range_start + ctx.chunk_size - 1 if ctx.chunk_size else None
-            if range_end and ctx.data_len is not None and range_end >= ctx.data_len:
-                range_end = ctx.data_len - 1
+            if req_end is not None:
+                # we're not allowed to download outside Range
+                range_end = req_end
+            else:
+                range_end = range_start + ctx.chunk_size - 1 if ctx.chunk_size else None
+                if range_end and ctx.data_len is not None and range_end >= ctx.data_len:
+                    range_end = ctx.data_len - 1
```

Comment:
shouldn't this check happen even if req_end?

Reply:
I don't quite understand what it does...
Anyway it should be maybe